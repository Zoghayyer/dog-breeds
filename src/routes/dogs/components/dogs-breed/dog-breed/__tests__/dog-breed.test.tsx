import React from 'react';
import { shallow } from 'enzyme';
import DogBreed, { OwnProps } from '../dog-breed';

describe('(Container) DogBreed', () => {
  let component;
  const defaultProps: OwnProps = {
    dog: {
      alt: '',
      favorite: false,
      id: 0,
      url: '',
    },
    handleClickFavorite: () => ({}),
  };

  const shallowDogBreed = (extendedProps: Partial<OwnProps> = {}) => {
    component = shallow(<DogBreed {...defaultProps} {...extendedProps} />);
  };

  beforeEach(() => shallowDogBreed());

  it('renders DogsBreedView component', () => {
    expect(component).toMatchSnapshot();
  });

  describe('(Method) handleClick', () => {
    it('calls handleClickFavorite with the correct id', () => {
      const handleClickFavoriteSpy = jest.fn();
      const props = {
        ...defaultProps,
        dog: {
          ...defaultProps.dog,
          id: 2,
        },
        handleClickFavorite: handleClickFavoriteSpy,
      };
      component = new DogBreed(props);

      component.handleClick();

      expect(handleClickFavoriteSpy).toBeCalledWith(2);
    });
  });
});
