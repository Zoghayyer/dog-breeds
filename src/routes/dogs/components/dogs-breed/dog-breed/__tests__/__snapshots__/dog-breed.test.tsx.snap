// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`(Container) DogBreed renders DogsBreedView component 1`] = `
<div
  key="0-"
>
  <Image
    alt=""
    className="dogs-image"
    src=""
  />
  <Icon
    alt="white heart icon"
    className="heart-icon"
    icon="whiteHeartIcon"
    onClick={[Function]}
  />
</div>
`;
